### PR TITLE
Added logic to check if the BQ Dataset exists before trying to re-create in the BQ Execute plugin

### PR DESCRIPTION
**Pull request by fernst**

### CODE REVIEW

**Review by itsankit-google — DISMISSED** [review]
LGTM. Thanks for the quick fix :)

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 1]
```diff
@@ -119,9 +119,15 @@ public void run(ActionContext context) throws Exception {
     String tableName = config.getTable();
     String datasetProjectId = config.getDatasetProject();
     if (!Strings.isNullOrEmpty(datasetName) && !Strings.isNullOrEmpty(tableName)) {
-      BigQuerySinkUtils.createDataset(bigQuery, DatasetId.of(datasetProjectId, datasetName), config.getLocation(),
-                                      cmekKeyName, () -> String.format("Unable to create BigQuery dataset '%s.%s'",
-                                                                       datasetProjectId, datasetName));
+      DatasetId datasetId = DatasetId.of(datasetProjectId, datasetName);
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: would it be better to put those logic to `BigQuerySinkUtils.createDataset` to make the operation idempotent ?

Comment (fernst) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: We always call this method with the assumption that this check has been made

Comment (seanzhougoogle) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: does that mean we need to check everywhere before calling this method ?

Comment (seanzhougoogle) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: would it be better to do check in one single place ?

Comment (seanzhougoogle) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: also checking in single place will avoid future bugs like this one.

Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: +1 to Sean, a second thought is it the correct behaviour that the util class method depends on the caller?

Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: Discussed offline with @fernst to make that method back to private and add another util method `createDatasetIfNotExists`.

**Review by seanzhougoogle — APPROVED** [review]
Thank you Ankit & Fernando

**Pull request merged**